src/main/java/org/apache/sysml/hops/recompile/Recompiler.java

```diff
@@ -1220,11 +1220,11 @@ else if (pb instanceof ForProgramBlock) //includes ParFORProgramBlock
 			ForProgramBlock pbTmp = (ForProgramBlock)pb;	
 			ForStatementBlock sbTmp = (ForStatementBlock) pbTmp.getStatementBlock();
 			//recompile predicate
-			if( sbTmp!=null &&!(et==ExecType.CP && !OptTreeConverter.containsMRJobInstruction(pbTmp.getFromInstructions(), true, true)) )			
+			if( sbTmp.getFromHops() != null && sbTmp!=null &&!(et==ExecType.CP && !OptTreeConverter.containsMRJobInstruction(pbTmp.getFromInstructions(), true, true)) )
```

Comment:
Should the `sbTmp!=null` condition be moved to the left of `sbTmp.getFromHops()!=null` condition to avoid calling `getFromHops()` on a potentially null `sbTmp`? (same goes for lines 1225 and 1227)